File: src/main/java/org/aldo/api/data/dto/YearlyDetailDto.java
package org.aldo.api.data.dto;

import lombok.Data;

import java.util.List;
@Data
public class YearlyDetailDto {
    private String collaborationId;
    private SummaryProfessorDto professor;
    private Integer totalExpectedHours;
    private List<YearlyHoursDto> collaborationHoursYearly;
}

File: src/app/model/dto/YearlyDetailDto.ts
import {SummaryProfessorDto} from "./SummaryProfessorDto";
import {YearlyHoursDto} from "./YearlyHoursDto";

export class YearlyDetailDto {
    collaborationId: string;
    professor: SummaryProfessorDto;
    totalExpectedHours: number;
    collaborationHoursYearly: YearlyHoursDto[];

    constructor(data: any) {
      this.collaborationId = data.collaborationId;
      this.professor = new SummaryProfessorDto(data.professor);
      this.totalExpectedHours = data.totalExpectedHours;
      this.collaborationHoursYearly = data.collaborationHoursYearly
    }
}
